examples/preprocessing/plot_target_encoder.py

```diff
@@ -0,0 +1,224 @@
+            "preprocessor": name,
+            "rmse_test_mean": rmse_test_score.mean(),
+            "rmse_test_std": rmse_train_score.std(),
+            "rmse_train_mean": rmse_train_score.mean(),
+            "rmse_train_std": rmse_train_score.std(),
+        }
+    )
+
+
+for name, categorical_preprocessor in categorical_preprocessors:
+    preprocessor = ColumnTransformer(
+        [
+            ("numerical", "passthrough", numerical_features),
+            ("categorical", categorical_preprocessor, categorical_features),
+        ]
+    )
+    pipe = make_pipeline(
+        preprocessor, HistGradientBoostingRegressor(random_state=0, max_iter=max_iter)
+    )
+    evaluate_model_and_store(name, pipe)
+
+
+# %%
+# Native Categorical Feature Support
+# ----------------------------------
+# In this section, we build and evaluate a pipeline that uses native categorical
+# feature support in :class:`~sklearn.ensemble.HistGradientBoostingRegressor`,
+# which only supports up to 255 unique categories. In our dataset, the most of
+# the categorical features have more than 255 unique categories:
+n_unique_categories = df[categorical_features].nunique().sort_values(ascending=False)
+n_unique_categories
+
+# %%
+# To workaround the limitation above, we group the categorical features into
+# low cardinality and high cardinality features. The high cardinality features
+# will be target encoded and the low cardinality features will use the native
+# categorical feature in gradient boosting.
+high_cardinality_features = n_unique_categories[n_unique_categories > 255].index
+low_cardinality_features = n_unique_categories[n_unique_categories <= 255].index
+mixed_encoded_preprocessor = ColumnTransformer(
+    [
+        ("numerical", "passthrough", numerical_features),
+        (
+            "high_cardinality",
+            TargetEncoder(target_type="continuous"),
+            high_cardinality_features,
+        ),
+        (
+            "low_cardinality",
+            OrdinalEncoder(handle_unknown="use_encoded_value", unknown_value=-1),
+            low_cardinality_features,
+        ),
+    ],
+    verbose_feature_names_out=False,
+)
+
+# The output of the of the preprocessor must be set to pandas so the
+# gradient boosting model can detect the low cardinality features.
+mixed_encoded_preprocessor.set_output(transform="pandas")
+mixed_pipe = make_pipeline(
+    mixed_encoded_preprocessor,
+    HistGradientBoostingRegressor(
+        random_state=0, max_iter=max_iter, categorical_features=low_cardinality_features
+    ),
+)
+mixed_pipe
+
+# %%
+# Finally, we evaluate the pipeline using cross validation and record the results:
+evaluate_model_and_store("mixed_target", mixed_pipe)
+
+# %%
+# Plotting the Results
+# --------------------
+# In this section, we display the results for all the encoders. When evaluating the
+# predictive performance on the test set, dropping the categories perform the
+# worst and the target encoders performs the best.
```

Comment:
There is a problem in the HTML rendering of the first bullet item. It probably requires a blank line:

```suggestion
# worst and the target encoders performs the best. This can be explained
# as follows:
#
```